backend/app/api/endpoints/knowledge.py

```
@@ -767,3 +789,173 @@ def get_qa_history(
+    from app.services.summary_service import get_summary_service
+
+    summary_service = get_summary_service(db)
+    summary = await summary_service.get_kb_summary(kb_id, current_user.id)
+    return KnowledgeBaseSummaryResponse(kb_id=kb_id, summary=summary)
+
+
+@summary_router.post("/{kb_id}/summary/refresh")
+async def refresh_kb_summary(
+    kb_id: int,
+    background_tasks: BackgroundTasks,
+    current_user: User = Depends(security.get_current_user),
```

Ruff (0.14.10)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)